RavenDB-21454: Splitting `LicenseAttribute.ServerWideTasks` into `LicenseAttribute.ServerWideBackup` and `LicenseAttribute.ServerWideExternalReplication` properties

ArieSLV opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21454

### Additional description

We encountered an issue with configuring server-wide backup on a free cloud instance.

This aspect forces us to split `LicenseAttribute.ServerWideTasks` into `LicenseAttribute.ServerWideBackup` and `LicenseAttribute.ServerWideExternalReplication` properties.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- This change requires a documentation update.

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

ArieSLV committed:
RavenDB-21454: Splitting `LicenseAttribute.ServerWideTasks` into `LicenseAttribute.ServerWideBackup` and `LicenseAttribute.ServerWideExternalReplication` properties

ppekrol commented on src/Raven.Server/ServerWide/ClusterStateMachine.License.cs:
```diff
@@ -499,9 +499,15 @@ private static void AssertServerWideFor(ServerStore serverStore, LicenseAttribut
+
+            case LicenseAttribute.ServerWideExternalReplications:
+                if (serverStore.LicenseManager.LicenseStatus.HasServerWideExternalReplications == false)
+                    throw new LicenseLimitException(LimitType.ServerWideExternalReplication, "Your license doesn't support adding external replication.");
```
server wide external replication

ArieSLV replied:
Done

kalczur commented:
I'll fix the studio build

kalczur committed:
RavenDB-21454 Use HasServerWideBackups and HasServerWideExternalReplications in studio

ArieSLV committed:
RavenDB-21454:
- Reorder new attributes to place them together
- Fix LimitType.ServerWideExternalReplications name for consistency

ppekrol commented on src/Raven.Server/ServerWide/ClusterStateMachine.License.cs:
```diff
@@ -499,9 +499,15 @@ private static void AssertServerWideFor(ServerStore serverStore, LicenseAttribut
+
+            case LicenseAttribute.ServerWideExternalReplications:
+                if (serverStore.LicenseManager.LicenseStatus.HasServerWideExternalReplications == false)
+                    throw new LicenseLimitException(LimitType.ServerWideExternalReplications, "Your license doesn't support adding external replication.");
```
server wide?

ArieSLV replied:
Done

ArieSLV committed:
RavenDB-21454: Exception message fix
